File: mysql_examples/recipes/select/date_sent.php
<?php
# date_sent.php - fetch rows into object, refer to columns by name

require_once "Cookbook.php";

$conn =& Cookbook::connect ();
if (PEAR::isError ($conn))
  die ("Cannot connect to server: " . $conn->getMessage ());

#@ FRAG
$result =& $conn->query ("SELECT srcuser,
                          DATE_FORMAT(t,'%M %e, %Y') AS date_sent
                          FROM mail");
if (PEAR::isError ($result))
  die ("Oops, the statement failed: " . $result->getMessage ());
while ($row =& $result->fetchRow (DB_FETCHMODE_OBJECT))
  printf ("user: %s, date sent: %s\n", $row->srcuser, $row->date_sent);
$result->free ();
#@ FRAG

$conn->disconnect ();

?>

File: mysql_examples/recipes/select/date_sent.py
#!/usr/bin/python
# date_sent.py - fetch rows into dictionary, refer to columns by name

import Cookbook
import MySQLdb

conn = Cookbook.connect ()

#@ _FRAG_
cursor = conn.cursor (MySQLdb.cursors.DictCursor)
cursor.execute ("""
                SELECT srcuser,
                DATE_FORMAT(t,'%M %e, %Y') AS date_sent
                FROM mail
                """)
for row in cursor.fetchall ():
  print "user: %s, date sent: %s" \
           % (row["srcuser"], row["date_sent"])
cursor.close ()
#@ _FRAG_

conn.close ()
